src/ts/components/product/ProductInfo.ts

```diff
@@ -0,0 +1,62 @@
+import { ProductInfoProps, ProductProps, ValidateProductProps, ProductEditProps } from "../../utils/interface";
+import { validateProductName, validateProductPrice, valudateProductQuantity, validateSameProductName } from "../../utils/validation";
+
+class ProductInfo implements ProductInfoProps{
+  private productList: ProductProps[] | null;
+
+  constructor() {
+    this.productList = this.getProductList();
+  }
+
+  validateProductInfo({ productName, productPrice, productQuantity }: ProductProps) {
+    const productNameList = this.productList.map((product: ProductProps) => product.productName);
+
+    validateProductName(productName);
+    validateProductPrice(+productPrice);
+    valudateProductQuantity(+productQuantity);
+    validateSameProductName(productName, productNameList);
+  }
+
+  validateEditProductInfo({ productName, productPrice, productQuantity, beforeProductName }: ValidateProductProps) {
+    const productNameList = this.productList
+      .map((product: ProductProps) => product.productName)
```

Comment:
```suggestion
      .map((product) => product.productName)
```
productList가 ProductProps[] 타입이라서 map을 사용할 때 인자에 명시하지 않더라도 ProductProps 타입으로 들어갑니다~